docs/providers.rst

```diff
@@ -0,0 +1,111 @@
+Providers for GCP-related Services
+==================================
+
+.. TODO: externally link "interfaces" to gordon-dns's docs once published.
+
+This package provides the following interface providers. Use any or all of them; they are not dependent on each other.
+
+
+Provider APIs
+-------------
+
+.. toctree::
+   :maxdepth: 1
+
+   compute
+   dns
+   pubsub
+
+
+Example Gordon Configuration with GCP Plugin
+--------------------------------------------
+
+.. literalinclude:: ../gordon.toml.example
+   :language: ini
+
+
+Plugin Configuration
+--------------------
+
+.. _gcp:
```

Comment:
These are custom references. Instead of saying `:ref:[gcp]` (and maybe needing to escape some backticks) I can just say `:ref:gcp` when I want to link here (which is done below).